packages/engine-server/src/markdown/remark/utils.ts

```diff
@@ -315,22 +316,24 @@ const getLinkCandidates = ({
         column: 1,
       };
     }
-    value.split(/\s+/).filter((word) => {
-      const maybeNote = notesMap.get(word);
-      if (maybeNote !== undefined) {
+    value.split(/\s+/).forEach((word) => {
+      const maybeNote = NoteUtils.getNotesByFnameFromEngine({
+        fname: word,
+        engine,
+      }).filter((note) => note.stub !== true);
+      if (!_.isEmpty(maybeNote)) {
         const candidate = {
           type: "linkCandidate",
           from: NoteUtils.toNoteLoc(note),
           value: value.trim(),
           position: textNode.position as Position,
           to: {
             fname: word,
-            vaultName: VaultUtils.getName(maybeNote.vault),
+            vaultName: VaultUtils.getName(maybeNote[0].vault),
```

Comment:
I just realized the past implementation here was mapping the notes wrong:

`notesMap` would overwrite existing keys as it was creating the fname / note maps.
So _technically_ `to.vaultName` should be the vault name of the _last_ element of `maybeNote` if we want to preserve the original behavior.

But since both are wrong and not accounting for same-name notes that exists in multiple vaults, I'll just take this up as a next item to fix it rather than creep in out-of-scope changes here.

Reply:
I actually went ahead and added a fix here that instead adds backlink candidates for all matching notes instead since it's a simple change.